core/src/test/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroupTest.java

```diff
@@ -19,21 +19,66 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.awaitility.Awaitility.await;
 
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Function;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.google.common.collect.ImmutableList;
 
 import com.linecorp.armeria.client.Endpoint;
 import com.linecorp.armeria.client.endpoint.DynamicEndpointGroup;
 import com.linecorp.armeria.common.util.AsyncCloseable;
+import com.linecorp.armeria.testing.junit.common.EventLoopExtension;
 
 class HealthCheckedEndpointGroupTest {
+
+    @RegisterExtension
+    static final EventLoopExtension eventLoop = new EventLoopExtension();
+
+    @Test
+    void delegateUpdateCandidatesWhileCreatingHealthCheckedEndpointGroup() {
+        final MockEndpointGroup delegate = new MockEndpointGroup();
+        final CompletableFuture<List<Endpoint>> future = delegate.initialEndpointsFuture();
+        future.complete(ImmutableList.of(Endpoint.of("127.0.0.1", 8080), Endpoint.of("127.0.0.1", 8081)));
```

Comment:
Just to confirm with the old code, this test fails with the exception. If so then it's probably good. I was sort of expecting not to complete this future (so initialCheckFuture.join() blocks forever).

Reply:
Yeah, it failed before.
I have to complete the future because the thread should be at `forEach(ctx -> ctx.initialCheckFuture.join());`. So another thread updates the endpoints and triggers the `ConcurrentModificationException`. 😄